src/nova_ws_handler.erl

```diff
@@ -40,8 +40,23 @@ init(Req, State = #{mod := Mod, controller_data := ControllerData}) ->
             ?ERROR("Websocket handler ~p returned unkown result ~p", [Mod, Error]),
             Req1 = cowboy_req:reply(500, Req),
             {ok, Req1, State}
+    end;
+init(Req, State) ->
+    %% We have something that needs securing
+    case nova_security_plugin:pre_http_request(State, #{}) of
+        {ok, State0 = #{req := Req}} ->
+            %% Recurse with secure set to false. This is hacky, but we can live with that now
+            init(Req, State0#{secure => false});
+        {stop, State0 = #{req := Req}} ->
+            %% We have no way to exit other than marking this is a unauthrized attempt and close in
+            %% websocket_init/1 function
+            init(Req, State0#{should_terminate => true,
```

Comment:
Why don't we return 401 here?

Reply:
We can not do that according to the websocket behaviour from cowboy.

Reply:
I was wrong. After digging around in cowboy source I found out that their documentation is partially wrong and the init-callback actually _can_ return a "normal" response. Will update this.